tls/s2n_async_pkey.c

```diff
@@ -393,19 +398,43 @@ S2N_RESULT s2n_async_pkey_sign_apply(struct s2n_async_pkey_op *op, struct s2n_co
 
     struct s2n_async_pkey_sign_data *sign = &op->op.sign;
 
+    /* Perform signature validation only if feature is buy in */
+    RESULT_ENSURE_REF(conn->config);
+    if (conn->config->async_pkey_validation_mode == S2N_ASYNC_PKEY_VALIDATION_STRICT) {
+        RESULT_ENSURE(!(s2n_async_pkey_verify_signature(conn, sign->sig_alg,
+                                                &sign->digest_copy, &sign->signature) < 0), S2N_ERR_VERIFY_SIGNATURE);
```

Comment:
Why is your ensure condition inverted? Wouldn't it just be 
```suggestion
        RESULT_ENSURE(s2n_async_pkey_verify_signature(conn, sign->sig_alg,
                                                &sign->digest_copy, &sign->signature) == S2N_SUCCESS, S2N_ERR_VERIFY_SIGNATURE);
```

If you use s2n_result it's even cleaner:
```suggestion
        RESULT_ENSURE(s2n_result_is_ok(s2n_async_pkey_verify_signature(conn, sign->sig_alg,
                                                &sign->digest_copy, &sign->signature)), S2N_ERR_VERIFY_SIGNATURE);
```

Reply:
Thanks for recommendation. Will update this